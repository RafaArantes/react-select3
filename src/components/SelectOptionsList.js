import React from 'react'
import PropTypes from 'prop-types'

import classNames from 'classnames'
import isFunction from 'lodash/isFunction'
import isNil from 'lodash/isNil'
import { stopPropagation } from '../utils/events'


const SelectOptionsList = ({ highlighted, selected, formatter, options = [], onSelect }, { cssClassNamePrefix }) => {
  const optionsList = options.map((option) => {
    const { id, isHidden } = option;

    if (isHidden) {
      return null
    }

    let optionText = option.text;

    if (isFunction(formatter)) {
      optionText = formatter(option)
    }

    const isSelected = !isNil(selected) && selected === id;
    const optionClassName = classNames(`${cssClassNamePrefix}__option`, {
      [`${cssClassNamePrefix}__option--selected`]: isSelected,
      [`${cssClassNamePrefix}__option--highlighted`]: id === highlighted,
    });
    const onOptionSelect = isSelected ? null : onSelect.bind(null, id);

    return (
      <li
        className={optionClassName}
        data-id={id}
        key={id}
        onClick={stopPropagation(onOptionSelect)}
      >
        { optionText }
      </li>
    )
  });

  return (
    <ul className={`${cssClassNamePrefix}__options-list`}>
      { optionsList }
    </ul>
  )
};

SelectOptionsList.contextTypes = {
  cssClassNamePrefix: PropTypes.string,
};

SelectOptionsList.propTypes = {
  formatter: PropTypes.func,
  highlighted: PropTypes.string,
  onSelect: PropTypes.func.isRequired,
  options: PropTypes.array.isRequired,
  selected: PropTypes.string,
};

export default SelectOptionsList
