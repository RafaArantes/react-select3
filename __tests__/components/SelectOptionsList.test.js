import SelectOptionsList from '../../src/components/SelectOptionsList'


describe('SelectOptionsList', () => {
  it('Should render', () => {
    const component = shallow(
      <SelectOptionsList options={ mock.options } onSelect={ jest.fn() }/>,
      selectComponentContext
    )

    expect(component).toMatchSnapshot()
  })
})
