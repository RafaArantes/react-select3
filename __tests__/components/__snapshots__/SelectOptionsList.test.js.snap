// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectOptionsList Should render 1`] = `
<ul
  className="rs3__options-list"
>
  <li
    className="rs3__option"
    data-id={3}
    key="3"
    onClick={[Function]}
  >
    three
  </li>
  <li
    className="rs3__option"
    data-id={4}
    key="4"
    onClick={[Function]}
  >
    four
  </li>
  <li
    className="rs3__option"
    data-id={5}
    key="5"
    onClick={[Function]}
  >
    five
  </li>
  <li
    className="rs3__option"
    data-id={6}
    key="6"
    onClick={[Function]}
  >
    six
  </li>
  <li
    className="rs3__option"
    data-id={7}
    key="7"
    onClick={[Function]}
  >
    seven
  </li>
  <li
    className="rs3__option"
    data-id={8}
    key="8"
    onClick={[Function]}
  >
    eight
  </li>
  <li
    className="rs3__option"
    data-id={9}
    key="9"
    onClick={[Function]}
  >
    nine
  </li>
  <li
    className="rs3__option"
    data-id={10}
    key="10"
    onClick={[Function]}
  >
    ten
  </li>
  <li
    className="rs3__option"
    data-id={11}
    key="11"
    onClick={[Function]}
  >
    eleven
  </li>
  <li
    className="rs3__option"
    data-id={12}
    key="12"
    onClick={[Function]}
  >
    twelve
  </li>
</ul>
`;
